### ISSUE TITLE
How to set 0x88a8 (802.1ad) service tag for QinQ interface?

### ISSUE THREAD

**Issue by ccy** [opnsense/core#5893]
#5560 mention double tagged QinQ vlan is ready.

In 22.7 RC1, I create a QinQ interface:

```
vlan02: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
        options=80000<LINKSTATE>
        ether 52:54:00:81:af:7f
        groups: vlan
        vlan: 222 vlanproto: 802.1q vlanpcp: 0 parent interface: vtnet0
        media: Ethernet autoselect (10Gbase-T <full-duplex>)
        status: active
        nd6 options=29<PERFORMNUD,IFDISABLED,AUTO_LINKLOCAL>
qinq01: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1496
        description: OPT2
        options=80000<LINKSTATE>
        ether 52:54:00:81:af:7f
        inet 192.168.111.13 netmask 0xffffff00 broadcast 192.168.111.255
        groups: vlan
        vlan: 111 vlanproto: 802.1q vlanpcp: 0 parent interface: vlan02
        media: Ethernet autoselect (10Gbase-T <full-duplex>)
        status: active
        nd6 options=29<PERFORMNUD,IFDISABLED,AUTO_LINKLOCAL>
```
In above setting, the double tagged packets use EtherType of `0x8100` (802.1Q) for both outer and inner tag.

The standard 802.1ad should use `0x88a8` as outer tag and `0x8100` as inner tag.

I couldn't find in OPNsense UI to specify to utilize 802.1ad protocol or `0x88a8` as outer tag.

**Comment by fichtner** [comment]
This is a kernel limitation of the QinQ implementation.

**Comment by AdSchellevis** [comment]
something looks off here, manually I seem to be able to create the right protocol: 
```
root@OPNsense:~/core # ifconfig vlan create vlandev le2_vlan4 vlan 5 vlanproto 802.1ad
vlan13
root@OPNsense:~/core # ifconfig vlan13
vlan13: flags=8842<BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1496
	ether 00:0c:29:50:76:45
	groups: vlan
	vlan: 5 vlanproto: 802.1ad vlanpcp: 0 parent interface: le2_vlan4
	media: Ethernet autoselect
	status: active
	nd6 options=23<PERFORMNUD,ACCEPT_RTADV,AUTO_LINKLOCAL>
```

We probably have to set vlanproto explicitly

**Comment by AdSchellevis** [comment]
Ok, I don't think I can fix this instantly as there is a limitation in how to handle `vlanproto` in the kernel. When we create the parent interface, we're not sure if there are (future) children and changing the proto doesn't seem to work.

```
ifconfig le2_vlan4 vlanproto 802.1ad
```

results in :
```
le2_vlan4: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> metric 0 mtu 1500
	description: OPT2
	ether 00:0c:29:50:76:45
	vlan: 4 vlanproto: 802.1q vlanpcp: 0 parent interface: le2
	media: Ethernet autoselect
	status: active
	nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
```

**Comment by fichtner** [comment]
More upstream context here, maybe move this to src.git for now?

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=264014

**Comment by fichtner** [comment]
@AdSchellevis shouldn't we put the upstream label back?

**Comment by AdSchellevis** [comment]
@fichtner reading the commit message of https://github.com/freebsd/freebsd-src/commit/b82b8055ad44cac6ee1e1201cf1ea4dbdd20cacd , I think we can fix core (change the parents proto to `802.1ad` when there are children involved) with this https://github.com/opnsense/core/commit/e1d8b471d092273fcb779e345df530c86c62d73c after the `ifconfig` code lands in our base.

**Comment by fichtner** [comment]
outer vlan "ad", inner vlan "q"
